### ISSUE TITLE
Reset issue with audio

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microbit-foundation/micropython-microbit-v2-simulator#93]
```python
from microbit import *
import music

music.pitch(185, 1000)
reset()
```

Click Reset repeatedly and quickly. Eventually you hit a large block of unexpected silence. The simulator then recovers.

This in and of itself isn't very likely in practice but it seems worth understanding what's going on as it's quite unexpected and could be the source of other bugs.

**Comment by microbit-matt-hillsdon** [comment]
Something similar but without audio

```python
from microbit import *
import random

print(random.randint(0, 100))
reset()
```

doesn't show any gap in output.

**Comment by microbit-matt-hillsdon** [comment]
The actual reset call isn't needed, just the manual resets.

```python
from microbit import *
import music

music.pitch(185, 1000)
```

Alternatively, this reproduces without manual resets:

```python
from microbit import *
import music

music.pitch(185, 50)
reset()
```

**Comment by microbit-matt-hillsdon** [comment]
This is unrelated to #91 - it reproduced with and without that fix.

**Comment by microbit-matt-hillsdon** [comment]
This is fixed by the changes on #99. Closing.
